### DISCUSSION TITLE
Rag Pipeline and some other thoughts

### DISCUSSION THREAD

**Discussion by SebastianStehle** [microsoft/semantic-kernel#5344] [Ideas]
Hi,

do you have any ideas for a RAG pipeline? There are some steps that could be unified, e.g.

* Create a summary of the result before it is added to the context.
* Rerank the results using external APIs like Cohere.
* Rerank the results using another promp.
* Calculate the embedding could just be one step.
* Filter the results based on permissions (if not supported by the vector store).

**Comment by sophialagerkranspandey** [comment]
If you're trying to build a RAG, we recommend using AzureAI search. You can learn more here: https://learn.microsoft.com/en-us/azure/search/retrieval-augmented-generation-overview

**Reply by SebastianStehle** [reply]
I think even azure ai search does not cover everything and can be combined with rerankers.
